src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -46,6 +46,7 @@ PreReleaseRequest<I>::~PreReleaseRequest() {
   if (!m_shutting_down) {
     m_image_ctx.state->handle_prepare_lock_complete();
   }
+  delete m_on_finish;
```

Comment:
@badone I would go with Number: 2, as changing the code in the base class may affect the other implementations (in the derived classes).

@trociny I'm closing this PR now.